src/state/browserHistory/index.ts

```diff
@@ -29,6 +30,12 @@ export const useBrowserHistoryStore = createRainbowStore<BrowserHistoryStore>(
         return { recents: newRecents };
       });
     },
+
+    hasVisited: (url: string) => {
+      const state = get();
+      return state.recents.some((site: { url: string }) => site.url === url);
+    },
```

Comment:
```suggestion
    hasVisited: (url: string) => {
      const normalizedUrl = normalizeUrlForRecents(url);
      return get().recents.some(site => site.url === normalizedUrl);
    },
```

(you'll need to import `normalizeUrlForRecents`)

Reply:
Nice. Made this change.